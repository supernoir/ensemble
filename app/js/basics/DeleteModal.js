import React from 'react';
import { Button, Header, Modal } from 'semantic-ui-react';
import { withRouter } from 'react-router-dom';
import intl from 'react-intl-universal';
import PropTypes from 'prop-types';

/**
 * DeleteModal Class
 * Asks the user to confirm the deletion of an item
 */
class DeleteModal extends React.Component {
	render() {
		return (
			<Modal open={this.props.open} onClose={() => this.props.close()}>
				<Modal.Header>{intl.get('modal.delete-title')}</Modal.Header>
				<Modal.Content>
					<Modal.Description>
						<Header>{intl.get('modal.delete-content-header', { entity: this.props.entity, ref: this.props.item })}</Header>
						<p>{intl.get('modal.delete-content-body', { entity: this.props.entity })}</p>
					</Modal.Description>
				</Modal.Content>
				<Modal.Actions>
					<Button color="black" onClick={() => this.props.close()}>
						{intl.get('app.cancel')}
					</Button>
					<Button
						negative
						icon="checkmark"
						labelPosition="right"
						content={intl.get('modal.delete-button-delete')}
						onClick={() => {
							this.props.confirmDelete(this.props.item);
							this.props.close();
							this.props.history.push(this.props.target);
						}}
					/>
				</Modal.Actions>
			</Modal>
		);
	}
}

export default withRouter(DeleteModal);

DeleteModal.propTypes = {
	close        : PropTypes.func,
	open         : PropTypes.func,
	history      : PropTypes.object,
	confirmDelete: PropTypes.func,
	entity       : PropTypes.string,
	item         : PropTypes.string,
	target       : PropTypes.string
};