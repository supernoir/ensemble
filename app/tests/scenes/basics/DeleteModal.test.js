

import React from 'react';
import { shallow } from 'enzyme';

const testData = {
	close        : jest.fn(),
	open         : jest.fn(),
	confirmDelete: jest.fn(),
	entity       : 'Project',
	item         : 'TestProject',
	target       : '/projects'
};

// Import Components
import DeleteModal from '../../../js/basics/DeleteModal';

test('DeleteModal -- Snapshot Test', () => {
	const wrapper = shallow(<DeleteModal
		close={() => testData.getProjectTitle()}
		open={() => testData.getProjectTitle()}
		confirmDelete={() => testData.getProjectTitle()}
		entity={testData.entity}
		item={testData.item}
		target={testData.target}
	/>
	);
	expect(wrapper).toMatchSnapshot();
});
