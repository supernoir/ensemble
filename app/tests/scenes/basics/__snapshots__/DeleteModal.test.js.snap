// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteModal -- Snapshot Test 1`] = `
<Route>
  <Component />
</Route>
`;
